R/api_url_query.R

```diff
@@ -0,0 +1,88 @@
+#' Generate URL code for API data query
+#'
+#' @description
+#' Generates the query code to append to the base API query URL for use by query_dataset_get().
+#'
+#' @param indicators Indicators required as a string or vector of strings (required)
+#' @param time_periods Time periods required as a string ("period|code") or vector of strings
+#' @param geographic_levels Geographic levels required as a string or vector of strings
+#' @param locations Location code required as a string or vector of strings
+#' @param filter_items Filter items required as a string or vector of strings
+#'
+#' @return String containing data query string to append to GET data query URL
+#' @export
+#'
+#' @examples
+#' api_url_query("session_count")
+#' api_url_query(c("session_count", "session_percent"))
+api_url_query <- function(
+    indicators,
+    time_periods = NULL,
+    geographic_levels = NULL,
+    locations = NULL,
+    filter_items = NULL) {
+  if (is.null(indicators)) {
+    stop("The keyword indicators must be supplied")
+  }
+  # Create the appropriate query strings for each level provided
+  if (!is.null(time_periods)) {
+    query_time_periods <- eesyapi::parse_filter_in(time_periods, "time_periods")
+  }
+  if (!is.null(geographic_levels)) {
+    query_geographic_levels <- eesyapi::parse_filter_in(
```

Comment:
If I give it a level with a space, it's not parsing that properly yet, e.g.
```
> eesyapi::api_url_query("session_count", geographic_levels = c("National", "Parliamentary Constituency", "MAT"))
[1] "?geographicLevels.in=National%2CParliamentary Constituency%2CMAT&indicators=session_count"
```

Reply:
Similar with locations - expect you've fix these together in the parsing function

```
eesyapi::api_url_query("session_count", locations = c("xxxxxxxxx / 23421348219", "E1293402134", 23432))
[1] "?locations.in=xxxxxxxxx / 23421348219%2CE1293402134%2C23432&indicators=session_count"
```